HeishaMon/HeishaMon.ino

```diff
@@ -589,93 +589,93 @@ void decode_heatpump_data() {
+  float Cool_Delta = (float)data[94] - 128;
+  if ( actData["Cool_Delta"] != Cool_Delta ) {
+    actData["Cool_Delta"] = Cool_Delta;
+    sprintf(log_msg, "received temperature (Cool_Delta): %.2f", Cool_Delta); log_message(log_msg);
+    sprintf(mqtt_topic, "%s/%s", mqtt_topic_base, "Cool_Delta"); mqtt_client.publish(mqtt_topic, String(Cool_Delta).c_str(), MQTT_RETAIN_VALUES);
   }
 
-  float TankHeatDelta = (float)data[99] - 128;
-  if ( actData["TankHeatDelta"] != TankHeatDelta ) {
-    actData["TankHeatDelta"] = TankHeatDelta;
-    sprintf(log_msg, "received temperature (TankHeatDelta): %.2f", TankHeatDelta); log_message(log_msg);
-    sprintf(mqtt_topic, "%s/%s", mqtt_topic_base, "TankHeatDelta"); mqtt_client.publish(mqtt_topic, String(TankHeatDelta).c_str(), MQTT_RETAIN_VALUES);
+  float Tank_Heat_Delta = (float)data[99] - 128;
+  if ( actData["Tank_Heat_Delta"] != Tank_Heat_Delta ) {
+    actData["Tank_Heat_Delta"] = Tank_Heat_Delta;
+    sprintf(log_msg, "received temperature (Tank_Heat_Delta): %.2f", Tank_Heat_Delta); log_message(log_msg);
+    sprintf(mqtt_topic, "%s/%s", mqtt_topic_base, "Tank_Heat_Delta"); mqtt_client.publish(mqtt_topic, String(Tank_Heat_Delta).c_str(), MQTT_RETAIN_VALUES);
   }
 
-  float Econsum = ((float)data[193] - 1.0) * 200;
-  if ( actData["Econsum"] != Econsum ) {
-    actData["Econsum"] = Econsum;
-    sprintf(log_msg, "received Watt (Elec consume): %.2f", Econsum); log_message(log_msg);
-    sprintf(mqtt_topic, "%s/%s", mqtt_topic_base, "Econsum"); mqtt_client.publish(mqtt_topic, String(Econsum).c_str(), MQTT_RETAIN_VALUES);
+  float Energy_Consumtion = ((float)data[193] - 1.0) * 200;
+  if ( actData["Energy_Consumtion"] != Energy_Consumtion ) {
+    actData["Energy_Consumtion"] = Energy_Consumtion;
+    sprintf(log_msg, "received Watt (Energy_Consumtion): %.2f", Energy_Consumtion); log_message(log_msg);
+    sprintf(mqtt_topic, "%s/%s", mqtt_topic_base, "Energy_Consumtion"); mqtt_client.publish(mqtt_topic, String(Energy_Consumtion).c_str(), MQTT_RETAIN_VALUES);
   }
 
-  float Eproduce = ((float)data[194] - 1.0) * 200;
-  if ( actData["Eproduce"] != Eproduce ) {
-    actData["Eproduce"] = Eproduce;
-    sprintf(log_msg, "received Watt (Elec produced): %.2f", Eproduce); log_message(log_msg);
-    sprintf(mqtt_topic, "%s/%s", mqtt_topic_base, "Eproduce"); mqtt_client.publish(mqtt_topic, String(Eproduce).c_str(), MQTT_RETAIN_VALUES);
+  float Energy_Production = ((float)data[194] - 1.0) * 200;
+  if ( actData["Energy_Production"] != Energy_Production ) {
+    actData["Energy_Production"] = Energy_Production;
+    sprintf(log_msg, "received Watt (Energy_Production): %.2f", Energy_Production); log_message(log_msg);
+    sprintf(mqtt_topic, "%s/%s", mqtt_topic_base, "Energy_Productions"); mqtt_client.publish(mqtt_topic, String(Energy_Production).c_str(), MQTT_RETAIN_VALUES);
```

Comment:
Small typo - a S too much in Energy_Production

Reply:
good one